consensus/src/main/java/org/ethereum/beacon/consensus/spec/EpochProcessing.java

```diff
@@ -0,0 +1,1050 @@
+    }
+  }
+
+  /*
+    """
+    Process the slashings.
+    Note that this function mutates ``state``.
+    """
+   */
+  default void process_slashings(MutableBeaconState state) {
+    EpochNumber current_epoch = get_current_epoch(state);
+    List<ValidatorIndex> active_validator_indices =
+        get_active_validator_indices(state.getValidatorRegistry(), current_epoch);
+    Gwei total_balance = get_total_balance(state, active_validator_indices);
+
+    // Compute `total_penalties`
+    Gwei total_at_start = state.getLatestSlashedBalances().get(current_epoch.increment()
+        .modulo(getConstants().getLatestSlashedExitLength()));
+    Gwei total_at_end = state.getLatestSlashedBalances()
+        .get(current_epoch.modulo(getConstants().getLatestSlashedExitLength()));
+    Gwei total_penalties = total_at_end.minusSat(total_at_start);
+
+    /* for index, validator in enumerate(state.validator_registry):
+        if validator.slashed and current_epoch == validator.withdrawable_epoch - LATEST_SLASHED_EXIT_LENGTH // 2:
+            penalty = max(
+                get_effective_balance(state, index) * min(total_penalties * 3, total_balance) // total_balance,
+                get_effective_balance(state, index) // MIN_PENALTY_QUOTIENT
+            )
+            state.validator_balances[index] -= penalty */
+
+    for (ValidatorIndex index : state.getValidatorRegistry().size()) {
+      ValidatorRecord validator = state.getValidatorRegistry().get(index);
+      if (validator.getSlashed() &&
+          current_epoch.equals(validator.getWithdrawableEpoch()
+              .minus(getConstants().getLatestSlashedExitLength().half()))) {
+        Gwei effective_balance = get_effective_balance(state, index);
+        Gwei penalty = UInt64s.max(
+            effective_balance.times(UInt64s.min(total_penalties.times(3), total_balance).dividedBy(total_balance)),
+            effective_balance.dividedBy(getConstants().getMinPenaltyQuotient())
+        );
+        state.getValidatorBalances().update(index, balance -> balance.minusSat(penalty));
+      }
+    }
+  }
+
+  /*
+    def eligible(index):
+      validator = state.validator_registry[index]
+      # Filter out dequeued validators
+      if validator.withdrawable_epoch != FAR_FUTURE_EPOCH:
+          return False
+      # Dequeue if the minimum amount of time has passed
+      else:
+          return get_current_epoch(state) >= validator.exit_epoch + MIN_VALIDATOR_WITHDRAWABILITY_DELAY
+   */
+  default boolean eligible(BeaconState state, ValidatorIndex index) {
+    ValidatorRecord validator = state.getValidatorRegistry().get(index);
+    // Filter out dequeued validators
+    if (!validator.getWithdrawableEpoch().equals(getConstants().getFarFutureEpoch())) {
+      return false;
+    } else {
+      // Dequeue if the minimum amount of time has passed
+      return get_current_epoch(state).greaterEqual(
+          validator.getExitEpoch().plus(getConstants().getMinValidatorWithdrawabilityDelay()));
```

Comment:
we are getting overflow here.`FAR_FUTURE` which is usually exit epoch is highest possible number and we are adding `MinValidatorWithdrawabilityDelay` to it

Reply:
👍